### ISSUE TITLE
Several bugs related to custom emoji and emoji aliases

### ISSUE THREAD

**Issue by WesleyAC** [zulip/zulip-mobile#4363]
Instructions to reproduce:

1. Add a custom emoji named `ok`
2. React with `ok` to a message
3. React with `got_it` to the same message

Now, the display bug show below happens, and also when you try to react to any message, the reaction is not shown on the frontend until you exit the message list view and then go back to the message list view.

![ok_got_it](https://user-images.githubusercontent.com/5001092/103319188-d7f44c80-4a6b-11eb-9ee5-ca4200b5b8d1.jpeg)

**Comment by gnprice** [comment]
Weird. Thanks for the report!

To make sure I'm understanding correctly, and to unpack that screenshot a bit:
* There's a built-in emoji `ok`, and `got_it` as an alias of it.
* So you uploaded a custom emoji replacing `ok`, and then reacted with both `ok` (which I guess should mean the new one) and `got_it` (which should probably mean the old one.)
* The mobile reactions list now shows two different reactions -- one with two people, and one with one.
  * The two people are both you.
  * That reaction displays as a black square. (Is that what the custom emoji is supposed to look like, or is this one of the bug symptoms?)
  * The other one displays with the built-in `ok`/`got_it` glyph.
  * I'm guessing that if you go to that tab, it'll show you, once.

**Comment by gnprice** [comment]
The next thing I'd do to debug this is to add some debugging in `MessageReactionList`, which is the component that's showing here, and investigate what it sees in the data. Specifically:
* What's `aggregatedReactions`, i.e. the return value of `aggregateReactions`?
* What's `message.reactions`, which is the main input to that function?

**Comment by abhi0504** [comment]
@gnprice Can i work on this issue

**Comment by WesleyAC** [comment]
@gnprice Yup, that description of the issue is correct. The black square is the image that I uploaded.

IMO the most troubling part of this bug is that this prevents new reactions from showing up client-side when the user adds a reaction. Not sure if that bug is also on the `MessageReactionList` codepath or elsewhere, since I haven't looked at the code or error logs or anything.

@abhi0504 feel free to take this issue! i don't plan to work on it at the moment.

**Comment by abhi0504** [comment]
@WesleyAC thanks for your reply @gnprice can you assign me this issue

**Comment by gnprice** [comment]
@abhi0504 Yep, please go ahead and work on the issue :slightly_smiling_face: In general, when an issue is labeled "help wanted" and nobody's already taken it, feel free to claim it by just saying so on the issue thread.

> IMO the most troubling part of this bug is that this prevents new reactions from showing up client-side when the user adds a reaction.

Oh, I see! You'd mentioned this in the original description:

> and also when you try to react to any message with any reaction emoji, the reaction is not shown on the frontend until you exit the message list view and then go back to the message list view.

but apparently I glossed over that. I agree that is troubling.

Can you spell out your reproduction steps to observe that symptom? IIUC it's something like:
* Do steps 1-3 above.
4. You're now in a message list, where you just added the two reactions above. (Is that right?)
5. Add another reaction, to the same message or any other message.
6. Observe that the reaction doesn't appear.
7. Repeat steps 5 and 6 any number of times.
8. Navigate back out to the main nav, and back in to the message list.
9. Observe that the reactions you added before are now there.
10. Add another reaction.
11. Observe that it now appears promptly, as usual. (Is this right?)

**Comment by WesleyAC** [comment]
11 is not correct — the bug persists even if you go back to the main nav then navigate back to the message list that has the double-reaction. Other than that, those steps are correct.

It's unclear to me if this only happens to the person who added the emojis, or if it also happens to anyone who happens to look at the messagelist, I haven't bothered to make another czo account to test that.

**Comment by abhi0504** [comment]
@gnprice  I followed up above 3 steps but this problem isn't reproduced
